app/models/media_object.rb

```diff
@@ -569,7 +563,48 @@ def move_bulk documents, params
 
   end
 
+  def _update_permalink_and_dependents
+    ensure_permalink!
+    self.parts.each do |master_file| 
+      begin
+        updated = master_file.ensure_permalink!
+        master_file.save( validate: false ) if updated
+      rescue
+        # no-op
+        # Save is called (uncharacteristically) during a destroy.
+      end
+    end
+
+    unless self.descMetadata.permalink.include? self.permalink 
+      self.descMetadata.permalink = self.permalink
+    end
+  end
+  handle_asynchronously :_update_permalink_and_dependents
+  
+  def update_permalink_and_dependents
+    self._update_permalink_and_dependents
+  end
+
+  def _remove_bookmarks
```

Comment:
Wait, never mind.  The problem of doing this asynchronously (and the problem with this at all) is that session variables aren't making it into Ability and thus losing access allowed by external groups like LDAP.